### PR TITLE
Deng-590 Add owner to table inventory

### PR DESCRIPTION
**Pull request by wwyc**
[DENG-590](https://mozilla-hub.atlassian.net/browse/DENG-590):  To start deprecating tables, it is helpful to be able to query table owners.  This PR will add owners column extracted from metadata labels to table_inventory_v1 table.  Owners from metadata.yaml files was added to labels in #3637 .

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.

[DENG-590]: https://mozilla-hub.atlassian.net/browse/DENG-590?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_tables_inventory_v1/query.py [hunk 1]
```diff
@@ -44,7 +44,6 @@ def create_last_modified_tmp_table(date, project, tmp_table_name):
     datasets = list(client.list_datasets())
 
     for dataset in datasets:
```
Comment (wwyc) [inline sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_tables_inventory_v1/query.py:46]:
```suggestion
    for dataset in datasets:
    
```

File: sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_tables_inventory_v1/query.py [hunk 2]
```diff
@@ -69,6 +69,24 @@ def create_last_modified_tmp_table(date, project, tmp_table_name):
...
+            SELECT table_catalog,
+                 table_schema,
+                 table_name,
+                 ARRAY_AGG(value) AS owners
+            FROM
+                (SELECT table_catalog, table_schema, table_name,
+                  ARRAY(
+                    SELECT as STRUCT ARR[OFFSET(0)] key, ARR[OFFSET(1)] value
+                    FROM UNNEST(REGEXP_EXTRACT_ALL(option_value, r'STRUCT\(("[^"]+", "[^"]+")\)')) kv,
+                    UNNEST([STRUCT(SPLIT(REPLACE(kv, '"', ''), ', ') as arr)])
```
Comment (ANich) [inline sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_tables_inventory_v1/query.py:82]: This parsing made me curious so I took a look at the `TABLE_OPTIONS` view and a string array of structs is an interesting way to store these 😅 I wonder why this was the choice...

File: sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_tables_inventory_v1/query.py [hunk 3]
```diff
@@ -69,6 +69,24 @@ def create_last_modified_tmp_table(date, project, tmp_table_name):
...
+                  ) options
+                FROM `{source_project}.region-us.INFORMATION_SCHEMA.TABLE_OPTIONS`
+                WHERE option_name = 'labels'
```
Comment (ANich) [inline sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_tables_inventory_v1/query.py:85]: Would this work better as a CTE?

**Review by ANich — APPROVED** [review]
Left one suggestion, lgtm otherwise

File: sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_tables_inventory_v1/query.py (hunk 3 above)
Comment (wwyc) [reply sql/moz-fx-data-shared-prod/monitoring_derived/bigquery_tables_inventory_v1/query.py]: Thanks for the suggestion!  I've moved it out as a CTE.

**Pull request merged**